### DISCUSSION TITLE
Restricted of service while using CAS

### DISCUSSION THREAD

**Discussion by rifkimubarok** [casdoor/casdoor#2095] [Q&A] [answered]
how to restrict apps or service when using CAS protocol?

**Comment by hsluoyz** [comment]
@rifkimubarok what do you mean by `restrict apps or service`?

**Reply by rifkimubarok** [reply]
for example, I want to make an application for only certain domains, but in the casdoor feature only oauth can be limited, authentication via CAS cannot (so unauthorized domains can use the CAS service)

**Reply by hsluoyz** [reply]
@rifkimubarok what's domain here? Can you use Casdoor phrases? Or just paste screenshots. Show `in the casdoor feature only oauth can be limited`

**Reply by rifkimubarok** [reply]
<img width="1680" alt="service" src="https://github.com/casdoor/casdoor/assets/42726159/dd9a1232-f6ce-4268-a36f-0f1be2d21716">
@hsluoyz i mean this service. when i enter google.com, the authentication still continue until send ticket.

**Comment by hsluoyz** [comment]
@ISulimanov do you know what he is talking about? I just can't understand the question and what he wants :(

**Reply by rifkimubarok** [reply]
<img width="1680" alt="expected" src="https://github.com/casdoor/casdoor/assets/42726159/5ed6ba1b-6bed-4fad-af0c-300b68e6a79f">
my expected output when the domain not registered

**Reply by hsluoyz** [reply]
Can you use a link under: https://door.casdoor.com as an example, and what you got and what you expect?

**Reply by ISulimanov** [reply]
@hsluoyz 
As I understand @rifkimubarok mean something like "white list" for allowed domains that may use casdoor for auth.

**Reply by rifkimubarok** [reply]
@ISulimanov yes, that's what i mean

**Reply by hsluoyz** [reply]
Is it something similar to OAuth's `Redirect URLs` in application edit page? https://door.casdoor.com/applications/casbin/app-casnode

So we can re-use this setting and show error when CAS service param is not inside this `Redirect URLs list, right?

![image](https://github.com/casdoor/casdoor/assets/3787410/f4a870d7-8593-4f02-b4f9-1050677f0a7a)

**Reply by rifkimubarok** [reply]
exactly, thats what i want.

**Reply by hsluoyz** [reply]
OK, issue tracked here: https://github.com/casdoor/casdoor/issues/2099

**Reply by hsluoyz** [reply, marked as answer]
@rifkimubarok fixed in: https://github.com/casdoor/casdoor/pull/2118
